### ISSUE TITLE
How about add `find method` php-doc to models in _ide_helper_models

### ISSUE THREAD

**Issue by lokielse** [barryvdh/laravel-ide-helper#109]
``` php
 @method static \Illuminate\Database\Query\Builder|\User find($value) 
```

**Comment by brazenvoid** [comment]
+1

**Comment by amitaymolko** [comment]
@barryvdh find method does return static, but the IDE (PhpStorm) isn't reading that as a model object (User in this case)...
Any idea why?

**Comment by geocine** [comment]
I just want to know if anyone is working on this so we could make a pull request ?

**Comment by stygiansabyss** [comment]
I think this is a resurfaced problem in 5.3.  It doesn't recognize find as static or a normal method.  In both cases it cannot find a declaration to go to.

**Comment by stygiansabyss** [comment]
Btw, if you have a base model class that all of your models extend, you can add the following to it's doc block to solve this.

`@method static \Illuminate\Database\Query\Builder|$this find($value)`

**Comment by brazenvoid** [comment]
@stygiansabyss Nope, doesn't work in PHPStorm. Both first, get and find need an explicit php doc on the model or above the line of code where they are used

**Comment by stygiansabyss** [comment]
Odd.  It's been working great for me.

**Comment by pahan35** [comment]
@barryvdh issue can be closed. ATM it works fine in PhpStorm.

I also made a [PR to Laravel](https://github.com/laravel/framework/pull/24526) which adds support for related methods like **findOrFail** etc. Unfortunately it's included only from Laravel **5.7**

**Comment by mfn** [comment]
@barryvdh ack, this can be closed. `static` is recognized by PhpStorm correctly as well as static analyzers I believe.
